users/vieting/jobs/fairseq.py

```diff
@@ -109,37 +88,86 @@ def create_tsv_and_labels(self):
-                    file=ltr_out,
+                    file=ltr,
                 )
-                print(audio_trans, file=wrd_out)
-        
-        # close all files
-        valid_tsv.close()
-        valid_ltr.close()
-        valid_wrd.close()
-        
-        train_tsv.close()
-        train_ltr.close()
-        train_wrd.close()
+                print(audio_trans, file=wrd)
+
+        tsv.close()
+        ltr.close()
+        wrd.close()
+
+    def get_common_dir(self):
+        """
+        Returns the common directory of all audios given in the corpora.
+        """
+        common_dir = None
+        # iterate over all corpora
+        for corpus_path in self.corpus_paths:
+            corpus_object = corpus.Corpus()
+            corpus_object.load(corpus_path.get())
+            for segment in corpus_object.segments():
+                audio_path = segment.recording.audio
+                assert os.path.exists(audio_path), f"Path {audio_path} does not exist."
+                if common_dir is None:
+                    common_dir = os.path.dirname(audio_path)
+                else:
+                    common_dir = os.path.commonpath([common_dir, os.path.dirname(audio_path)])
+        return common_dir
+
 
+class MergeLabeledFairseqDataJob(Job):
+    """
+    Merge multiple labeled fairseq data directories into one. Each directory must contain the following files:
+    - <name>.tsv
+    - <name>.ltr
+    - <name>.wrd
 
+    If create_letter_dict is set to True, the following file will be created:
+    - dict.ltr.txt
+    """
+    def __init__(
+        self,
+        labeled_data_paths: Union[List[tk.Path], tk.Path],
+        create_letter_dict: bool = True,
+    ):
+        if not isinstance(labeled_data_paths, list):
+            labeled_data_paths = [labeled_data_paths]
+        self.labeled_data_paths = labeled_data_paths
+        assert all([isinstance(path, tk.Path) for path in self.labeled_data_paths])
+
+        self.create_letter_dict = create_letter_dict
+
+        self.out_task_path = self.output_path("task", directory=True)
+        self.out_dict_ltr_path = self.output_path("task/dict.ltr.txt")
+
+    def tasks(self):
+        yield Task("run", rqmt=self.rqmt, mini_task=True)
+
+    def run(self):
+        self.merge()
+        if self.create_letter_dict:
+            self.create_dict_ltr()
+
+    def merge(self):
+        for path in self.labeled_data_paths:
+            assert os.path.exists(path.get()), f"Path {path} does not exist."
+
+            # iterature through directory "path" and copy all files to self.out_task_path directory
+            for file in os.listdir(path.get()):
+                if file.endswith(".tsv") or file.endswith(".ltr") or file.endswith(".wrd"):
+                    src = os.path.join(path.get(), file)
+                    dst = os.path.join(self.out_task_path.get(), file)
+                    shutil.copyfile(src, dst)
```

Comment:
You might check that `dst` does not exist yet because it would be unexpected behavior if two input folders have files with the same name. I think it would be better to crash in that case than to ignore one of them silently.
```suggestion
                    assert not os.path.exists(dst), f"'{dst}' exists, two inputs have the same names"
                    shutil.copyfile(src, dst)
```